# Assignment-1

## Student details 

Name: Harshilkumar Vaghela,
 Student Number: 3116318

## Chpater-1

### Exercise 1.1

It is a good idea to read this book in front of a computer so you can try out the
examples as you go.
Whenever you are experimenting with a new feature, you should try to make mistakes. For example,
in the “Hello, world!” program, what happens if you leave out one of the quotation marks? What if
you leave out both? What if you spell print wrong?
This kind of experiment helps you remember what you read; it also helps when you are programming,
because you get to know what the error

In [6]:
pint('Hello, world!')

NameError: name 'pint' is not defined

1. In a print statement, what happens if you leave out one of the parentheses, or both?

In [7]:
print('Hello, world!

SyntaxError: EOL while scanning string literal (<ipython-input-7-b44f6a6ce21f>, line 1)

In [8]:
print 'Hello, world!'

SyntaxError: Missing parentheses in call to 'print'. Did you mean print('Hello, world!')? (<ipython-input-8-653b30cd70a8>, line 1)

2. If you are trying to print a string, what happens if you leave out one of the quotation marks,
or both?

In [2]:
print('Hello, world!)

SyntaxError: EOL while scanning string literal (<ipython-input-2-9739917fae95>, line 1)

In [5]:
print(Hello, world!)

SyntaxError: invalid syntax (<ipython-input-5-d14e6f3d0ea2>, line 1)

3. You can use a minus sign to make a negative number like -2. What happens if you put a plus
sign before a number? What about 2++2?

In [9]:
2+-2

0

In [10]:
2++2

4

4. In math notation, leading zeros are ok, as in 09. What happens if you try this in Python?
What about 011?

In [11]:
011

SyntaxError: invalid token (<ipython-input-11-2e9780ecac58>, line 1)

5. What happens if you have two values with no operator between them?

In [13]:
2 2

SyntaxError: invalid syntax (<ipython-input-13-ed76b62ef94d>, line 1)

### Exercise 1.2

Start the Python interpreter and use it as a calculator.

1. How many seconds are there in 42 minutes 42 seconds?

In [13]:
42*60+42 #60 seconds in a minute

2562

2562 seconds

2. How many miles are there in 10 kilometers? Hint: there are 1.61 kilometers in a mile.

In [16]:
10/1.61

6.211180124223602

6.211180124223602 miles

3. If you run a 10 kilometer race in 42 minutes 42 seconds, what is your average pace (time per
mile in minutes and seconds)? What is your average speed in miles per hour?

In [4]:
2562 / 6.211180124223602 #time in seconds/distance in miles

412.482

In [3]:
412.482 / 60 #time in minutes/distance in miles

6.874700000000001

Average pace=412.482 seconds/mile=6.874700000000001 minutes/mile

In [6]:
((10/1.61)/(42*60 +42))*3600#(distance in miles/time in seconds)multiplied by 3600(3600 seconds in hour)

8.727653570337614

average speed is 8.727653570337614 miles/hour

## Chapter 2

### Exercise 2.1

Repeating my advice from the previous chapter, whenever you learn a new feature,
you should try it out in interactive mode and make errors on purpose to see what goes wrong.

• We’ve seen that n = 42 is legal. What about 42 = n?

In [32]:
42=n

SyntaxError: can't assign to literal (<ipython-input-32-94a448c80b86>, line 1)

• How about x = y = 1?

In [33]:
x=y=1

• In some languages every statement ends with a semi-colon, ;. What happens if you put a
semi-colon at the end of a Python statement?

In [6]:
x=2;

• What if you put a period at the end of a statement?

In [7]:
x=3.

• In math notation you can multiply x and y like this: xy. What happens if you try that in
Python?

In [5]:
xy

NameError: name 'xy' is not defined

### Exercise 2.2

Practice using the Python interpreter as a calculator:

1. The volume of a sphere with radius r is 4*pi*(cube of radius). What is the volume of a sphere with radius 5?

In [40]:
pi = 3.1415926535897932
n=4/3
radius=5
n*pi*(radius**3)

523.5987755982989

2. Suppose the cover price of a book is 24.95 dollars, but bookstores get a 40% discount. Shipping costs
3 dollars for the first copy and 75 cents for each additional copy. What is the total wholesale cost for
60 copies?

In [7]:
bookPrice=((100-40)*24.95)/100#price for one book with a discount 
sixtyBookprice=bookPrice*60
totalShippingCost=3+(0.75*59)
totalWholesalePrice=totalShippingCost+sixtyBookprice
totalWholesalePrice

945.45

945 dollars and 45 cents is the total wholesale price for sixty books

3. If I leave my house at 6:52 am and run 1 mile at an easy pace (8:15 per mile), then 3 miles at
tempo (7:12 per mile) and 1 mile at easy pace again, what time do I get home for breakfast?

In [9]:
start_time_seconds=(6*60+52)*60 #converting the time of the day in seconds for easiness
easy_time_seconds=(8*60+15)*2 #assigning time he took in running at easy pace in seconds(note:he ran two hours of his total run at easy pace.)
tempo_time_seconds=(7*60+12)*3 #just the same as easy pace for tempo
totalTime_hours=(start_time_seconds+easy_time_seconds+tempo_time_seconds)/3600
totalTime_hours

7.501666666666667

In [15]:
0.5016666666666678*60 #converting the value in hours after decimal point in minutes 

30.10000000000007

Ignoring the value after decimal point in minutes, he got back at home for breakfast at 7:30 am.

## Reason for joining this course 

I have a basic knowledge of java as I have studied the course programming-1 at this university, I am taking this course to learn where python is used in computer science field, how is it differnt from java and to know about the basic use of python language. 